### ISSUE TITLE
Part A: Stacks (40 points)

### ISSUE THREAD

**Issue by karlnicholas** [CS111C-META-FALL-2016/LabE-Team1#3]
* Implement a new kind of stack that only accepts a single copy of an object in the stack (i.e., it does not allow duplicates). 
    * If an element is pushed onto the stack, and the element already exists in the stack, the stack will remain unchanged.
  * Include a method moveToTop that allows you to prioritize an element in the stack. 
    * If invoked with an element currently in the stack, the method moves the element to the top of the stack.
    * If invoked with an element not in the stack, the method adds the element to the top of the stack (as would normally happen with push).
  * Your class extends ArrayStack and implements NoDupsPrioritizeStackInterface (provided). 
    * The modified ArrayStack class has protected variables so you can directly access them in NoDupsPrioritizeArrayStack.
    * In addition to implementing the two new methods moveToTop and display, you must override push so that duplicates are not added.

**Comment by karlnicholas** [comment]
Working version of NoDupsPrioritizeArrayStack posted. Please review.

**Comment by jillHuang7** [comment]
It is compiled and passed the test!

**Comment by karlnicholas** [comment]
I need to ask you to make a more definitive code review, as you will see that I did with your code. I know it compiled and passed the test because I checked it in and asked for a code review. On the job you will be expected to do code reviews and it looks bad it you don't spend the time to look at the code carefully. Not only will you learn from the exercise, but if you find a bug it will very helpful to the whole team. It's important that the coders find the bugs before the customers do.
